### PR TITLE
Add initial tests for /github/:project

### LINKED ISSUE #3331: Add tests for `/github/:project` in `dependency-discovery`
In #3295, we added the new path `/github/:project-name` (also `/github/:namespace/:project-name`) into the `dependency-discovery` service to give support for searching GitHub issues.

In the same spirit of https://github.com/Seneca-CDOT/telescope/issues/2828, we would like to add tests for this route, too.

### PR DESCRIPTION
**Pull request by JerryHue**
## Issue This PR Addresses

Fixes #3331

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI
- [ ] **Test**: Change that adds a new test case

## Description

This PR adds some test cases to the route `/github/:project/` in the `dependency-discovery`.

## Steps to test the PR

I don't think you need to test this locally, just let CI do the checks for you 😉 

However, if you insist:

1. Get my changes,
2. Run the tests for only the service so that it is faster (`pnpm run jest dependency.test.js`).

## Checklist

- [X] **Quality**: This PR builds and passes our npm test and works locally
- [X] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: src/api/dependency-discovery/test/dependency.test.js [hunk 1]
```diff
@@ -28,3 +54,42 @@ describe('GET /projects', () => {
...
+    expect(res.status).toBe(200);
+    expect(res.body).toEqual(gitHubIssuesForDep);
+  });
+
+  test('Should return 200 and an object with the GitHub issues if dependency name has namespace', async () => {
+    const dependencyName = '@babel/code-frame';
+    const gitHubIssuesForDep = depsGitHubIssuesList[dependencyName];
+    const res = await request(app).get(`/github/${dependencyName}`);
+    expect(res.status).toBe(200);
+    console.log(res.body);
```
Comment (humphd) [inline src/api/dependency-discovery/test/dependency.test.js:77]: You can remove this

File: src/api/dependency-discovery/test/dependency.test.js [hunk 2]
```diff
@@ -28,3 +54,42 @@ describe('GET /projects', () => {
...
+    expect(res.body).toEqual(gitHubIssuesForDep);
+  });
+
+  test('Should return 404 if dependency does not exist', async () => {
+    const dependencyName = 'this-is-not-a-dependency';
+    const res = await request(app).get(`/github/${dependencyName}`);
+    console.log(res.body);
```
Comment (humphd) [inline src/api/dependency-discovery/test/dependency.test.js:84]: Same there, remove this.

**Review by humphd — CHANGES_REQUESTED** [review]
Looks good, just remove the console.logs

**Review by dbelokon — DISMISSED** [review]
Nice! Looks good to me!

**Review by humphd — APPROVED** [review]

**Review by tcvan0707 — APPROVED** [review]

**Review by dbelokon — APPROVED** [review]
LGTM

**Pull request merged**